### ISSUE TITLE
Support passing ArrayBuffer to Filer.Buffer() constructor

### ISSUE THREAD

**Issue by humphd** [filerjs/filer#249]
A common pattern for browser-based Filer work is to XHR some binary data and then want to store it in the filesystem.  XHR2 allows you to specify that you want to get the binary data as an `'arraybuffer'`, which then needs to get used via a typed array.

Node's `Buffer` can take a typed array but not an `ArrayBuffer` in the constructor.  And, the browserify buffer implementation we use is true to this interface (i.e., it silently throws away the `ArrayBuffer` just like node does).  As such, I can't imagine them taking a patch upstream for this.

However, it would be nice for us to do it in `Filer.Buffer`.  Basically it means that we inspect the arg passed to `Filer.Buffer`, and if it's an `ArrayBuffer`, we wrap it in a `Uint8Array` and pass that along.  The nice thing about browserify's buffer is that it will decorate the `Uint8Array` you pass to it, so there is no copy operation.

Basically, we want to allow one to do either of these, instead of only the first:

``` js
// given `data`, an ArrayBuffer returned from an XHR get 
var buf1 = new Filer.Buffer(new Uint8Array(data));

var buf2 = new Filer.Buffer(data);
```

cc @secretrobotron, @alicoding
